decompiler/backend/codevisitor.py

```diff
@@ -48,15 +48,14 @@ def visit_seq_node(self, node: ast_nodes.SeqNode) -> str:
     def visit_loop_node(self, node: ast_nodes.LoopNode) -> str:
         """Generate code for loops."""
         loop_after = ""
-        if node.is_endless_loop:
-            loop_type = "while (true)"
-        elif isinstance(node, ast_nodes.DoWhileLoopNode):
```

Comment:
The code transforms every endless loop into a `while(true)` loop and conflicts with the `force_for_loop` option.

Also do you have an example, where a `do{...}while(true)` will be generated? 
All my examples (even with assembly) were optimized to a `while(true)`  loop

Reply:
No, in general, each endless loop should be transformed into an endless-while loop. But I am not sure what happens if we start to simplify conditions of do-while loops at some point in the code generator. Here, it could happen that may be one will become an endless loop. We can also say, it does not happen, and if so we consider it again. This is me being paranoid.

Reply:
Readded the code, but it also checks if the node is a dowhileloop. 
Only then a while(true) will be generated. Other endlessloops (e.g. while/for) will be transformed normally. 
==> no conflict with `force_for_loops` + no chance of an endless do while